### ISSUE TITLE
Unable to publish to Artifactory starting from version 0.18

### ISSUE THREAD

**Issue by lombarde** [arktekk/sbt-aether-deploy#44]
I started to use your plugin to deploy on Artifactory. While with version <= 0.17 things are working fine, with the version >= 0.18 I am getting this error while running 'sbt publish':

Service of type interface org.eclipse.aether.impl.VersionResolver failed to be crated by impl type class org.apache.maven.repository.internal.DefaultVersionResolver
java.lang.NoClassDefFoundError: org/apache/commons/lang3/Validate
	at org.apache.maven.repository.internal.DefaultVersionResolver.setMetadataResolver(DefaultVersionResolver.java:140)
	at org.apache.maven.repository.internal.DefaultVersionResolver.initService(DefaultVersionResolver.java:121)
	at org.eclipse.aether.impl.DefaultServiceLocator$Entry.newInstance(DefaultServiceLocator.java:171)
	at org.eclipse.aether.impl.DefaultServiceLocator$Entry.getInstances(DefaultServiceLocator.java:141)
	at org.eclipse.aether.impl.DefaultServiceLocator$Entry.getInstance(DefaultServiceLocator.java:127)
	at org.eclipse.aether.impl.DefaultServiceLocator.getService(DefaultServiceLocator.java:291)
	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.initService(DefaultRepositorySystem.java:139)
	at org.eclipse.aether.impl.DefaultServiceLocator$Entry.newInstance(DefaultServiceLocator.java:171)
	at org.eclipse.aether.impl.DefaultServiceLocator$Entry.getInstances(DefaultServiceLocator.java:141)
	at org.eclipse.aether.impl.DefaultServiceLocator$Entry.getInstance(DefaultServiceLocator.java:127)
	at org.eclipse.aether.impl.DefaultServiceLocator.getService(DefaultServiceLocator.java:291)
	at aether.internal.Booter$.newRepositorySystem(Booter.scala:48)
	at aether.internal.Booter$.apply(Booter.scala:54)
	at aether.AetherPlugin$class.deployIt(Plugin.scala:123)
	at aether.AetherPlugin$.deployIt(Plugin.scala:27)
	at aether.AetherPlugin$$anonfun$deployTask$1.apply(Plugin.scala:74)
	at aether.AetherPlugin$$anonfun$deployTask$1.apply(Plugin.scala:72)
	at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
	at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
	at sbt.std.Transform$$anon$4.work(System.scala:63)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
	at sbt.Execute.work(Execute.scala:237)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
	at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.ClassNotFoundException: org.apache.commons.lang3.Validate
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 34 more
Service of type interface org.eclipse.aether.RepositorySystem failed to be crated by impl type class org.eclipse.aether.internal.impl.DefaultRepositorySystem
java.lang.IllegalArgumentException: version resolver has not been specified
	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.setVersionResolver(DefaultRepositorySystem.java:162)
	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.initService(DefaultRepositorySystem.java:139)
	at org.eclipse.aether.impl.DefaultServiceLocator$Entry.newInstance(DefaultServiceLocator.java:171)
	at org.eclipse.aether.impl.DefaultServiceLocator$Entry.getInstances(DefaultServiceLocator.java:141)
	at org.eclipse.aether.impl.DefaultServiceLocator$Entry.getInstance(DefaultServiceLocator.java:127)
	at org.eclipse.aether.impl.DefaultServiceLocator.getService(DefaultServiceLocator.java:291)
	at aether.internal.Booter$.newRepositorySystem(Booter.scala:48)
	at aether.internal.Booter$.apply(Booter.scala:54)
	at aether.AetherPlugin$class.deployIt(Plugin.scala:123)
	at aether.AetherPlugin$.deployIt(Plugin.scala:27)
	at aether.AetherPlugin$$anonfun$deployTask$1.apply(Plugin.scala:74)
	at aether.AetherPlugin$$anonfun$deployTask$1.apply(Plugin.scala:72)
	at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
	at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
	at sbt.std.Transform$$anon$4.work(System.scala:63)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
	at sbt.Execute.work(Execute.scala:237)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
	at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
java.lang.RuntimeException: Failed to create RepositorySystem. This cannot be good!
	at scala.sys.package$.error(package.scala:27)
	at aether.internal.Booter$.newRepositorySystem(Booter.scala:49)
	at aether.internal.Booter$.apply(Booter.scala:54)
	at aether.AetherPlugin$class.deployIt(Plugin.scala:123)
	at aether.AetherPlugin$.deployIt(Plugin.scala:27)
	at aether.AetherPlugin$$anonfun$deployTask$1.apply(Plugin.scala:74)
	at aether.AetherPlugin$$anonfun$deployTask$1.apply(Plugin.scala:72)
	at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
	at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
	at sbt.std.Transform$$anon$4.work(System.scala:63)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
	at sbt.Execute.work(Execute.scala:237)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
	at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
java.lang.RuntimeException: Failed to create RepositorySystem. This cannot be good!
	at scala.sys.package$.error(package.scala:27)
	at aether.internal.Booter$.newRepositorySystem(Booter.scala:49)
	at aether.internal.Booter$.apply(Booter.scala:54)
	at aether.AetherPlugin$class.deployIt(Plugin.scala:123)
	at aether.AetherPlugin$.deployIt(Plugin.scala:27)
	at aether.AetherPlugin$$anonfun$deployTask$1.apply(Plugin.scala:74)
	at aether.AetherPlugin$$anonfun$deployTask$1.apply(Plugin.scala:72)
	at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
	at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
	at sbt.std.Transform$$anon$4.work(System.scala:63)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
	at sbt.Execute.work(Execute.scala:237)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
	at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[error] (*:aetherDeploy) Failed to create RepositorySystem. This cannot be good!
[error] Total time: 0 s, completed May 4, 2017 1:51:42 PM

**Comment by drieks** [comment]
I have the same problem, I added

```
libraryDependencies += "org.apache.commons" % "commons-lang3" % "3.1"
```

to `project/plugins.sbt` as a workaround

**Comment by hamnis** [comment]
Should be fixed in 0.19.0
